src/server/scripts/Outland/zone_shadowmoon_valley.cpp

```diff
@@ -571,64 +589,103 @@ class npc_dragonmaw_peon : public CreatureScript
     {
         npc_dragonmaw_peonAI(Creature* creature) : ScriptedAI(creature) { }
 
+        EventMap events;
+        EventMap events2;
         uint64 PlayerGUID;
-        bool Tapped;
-        uint32 PoisonTimer;
 
         void Reset()
         {
+            events.Reset();
+            events2.Reset();
             PlayerGUID = 0;
-            Tapped = false;
-            PoisonTimer = 0;
+        }
+
+        void EnterCombat(Unit* /*who*/)
+        {
+            events.ScheduleEvent(EVENT_KICK, urand(5000, 10000));
+            events.ScheduleEvent(EVENT_SUNDER, urand(5000, 10000));
         }
 
         void SpellHit(Unit* caster, const SpellInfo* spell)
         {
             if (!caster)
                 return;
 
-            if (caster->GetTypeId() == TYPEID_PLAYER && spell->Id == 40468 && !Tapped)
-            {
-                PlayerGUID = caster->GetGUID();
+            PlayerGUID = caster->GetGUID();
 
-                Tapped = true;
-                float x, y, z;
+            if (caster->GetTypeId() == TYPEID_PLAYER && spell->Id == SPELL_POISON)
+            {
                 caster->GetClosePoint(x, y, z, me->GetObjectSize());
-
-                me->SetWalk(false);
-                me->GetMotionMaster()->MovePoint(1, x, y, z);
+                Talk(SAY_1);
+                events2.ScheduleEvent(EVENT_WALK_TO_MUTTON, 0);
             }
         }
 
-        void MovementInform(uint32 type, uint32 id)
+        void MovementInform(uint32 /*type*/, uint32 id)
         {
-            if (type != POINT_MOTION_TYPE)
-                return;
+            if (id == 1)
+            {
+                me->HandleEmoteCommand(EMOTE_ONESHOT_EAT);
+                events2.ScheduleEvent(EVENT_POISIONED, 5000);
+            }
+        }
 
-            if (id)
+        void CreditPlayer()
+        {
+            if (PlayerGUID)
             {
-                me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_EAT);
-                PoisonTimer = 15000;
+                Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID);
```

Comment:
//Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID);
				if (Player* player = SelectTargetFromPlayerList(100.0f))

Reply:
so if i add this in, if a player who has the quest in 100.0f it will reward them too? instead of the correct way as i have done?

Reply:
Because the task cannot be completed now